### PR TITLE
Add air_temperature

### PR DESCRIPTION
**Pull request by maximlt**
The `air_temperature` xarray dataset that we usually get from `xarray.tutorial.open_dataset('air_temperature')` is used quite often in hvPlot, also in GeoViews and HoloViews. We discussed with @Azaya89 shipping a smaller version, as the original NetCDF file is pretty big, around 7MB. This PR adds this reduced version that is a subset of the original one with only 20 timestamps (5 days of data, one timestamp every 6 hours) instead of 2920 timestamps (2 years).

The dataset was created by running this script:

```python
from pathlib import Path
import xarray as xr
import hvplot.xarray

air_temp = xr.tutorial.load_dataset('air_temperature')
air_temp_small = air_temp.sel(time=slice("2014-02-24", "2014-02-28"))
of = Path('/tmp/air_temperature_small.nc')
air_temp_small.to_netcdf(of)
```

With these dependencies:
```
# Name                    Version                   Build  Channel
libgdal-netcdf            3.10.0               hd29f476_2    conda-forge
libnetcdf                 4.9.2           nompi_he469be0_114    conda-forge
netcdf4                   1.7.1           nompi_py312hd88a8d4_102    conda-forge
rioxarray                 0.17.0             pyhd8ed1ab_0    conda-forge
xarray                    2024.11.0          pyhd8ed1ab_0    conda-forge
```

The resampled file weighs 70KB. We haven't yet defined a global size limit for hvsampledata or an individual file size limit, but I'd guess this is small enough to be accepted.

The date chosen was obtained after a quick analysis, attempting to find the timestamps that showed the most spatial and temporal variations (to avoid picking the most boring part).

```python
da = air_temp.air
# Spatial variation per timestamp
spatial_std = da.std(dim=["lat", "lon"])
# Temporal variation
temporal_gradient = np.abs(da.mean(dim=["lat", "lon"]).diff(dim="time"))
# Normalize
spatial_std_norm = (spatial_std - spatial_std.min()) / (spatial_std.max() - spatial_std.min())
temporal_gradient_norm = (temporal_gradient - temporal_gradient.min()) / (temporal_gradient.max() - temporal_gradient.min())
# Combine with equal weights
variation_metric = spatial_std_norm[:-1] + temporal_gradient_norm
# Find the max over 20 timestamps
rolling_variation = variation_metric.rolling(time=20, center=True).sum()
argmax_index = rolling_variation.argmax(dim="time")
da.time[rolling_variation.argmax(dim="time")]
```

image

image

---

API-wise, I'm again (https://github.com/holoviz/hvsampledata/pull/23) tempted to make `engine` a required argument, even if I'm not sure we're ever going to support another library to read gridded dataset after `xarray`.

### CODE REVIEW

**Review by Azaya89 — APPROVED** [review]
Thank you. This LGTM!

**Comment by hoxbro** [comment]
I don't think we need to call it `air_temperature_small`; I would just call it `air_temperature`.

**Comment by maximlt** [comment]
> I don't think we need to call it `air_temperature_small`; I would just call it `air_temperature`.

It's possible we'll add to `hvsampledata` the full `air_temperature` dataset but not ship it? It could also be confusing to have a dataset named like the original xarray one but being in fact different? But I also see it's not the ideal name.

An example of a plot that wouldn't look good if built from the small version.
image

**Comment by maximlt** [comment]
Based on a discussion we had, name updated from `air_temperature_small` to `air_temperature`. Hopefully it won't cause any confusion!

File: scripts/generate_typehints.py [hunk 1]
```diff
@@ -46,14 +46,14 @@ def {func_name}(
 def generate_gridded_overloads(func_name: str) -> str:
     """Generates overloads for a function with the specified engines and lazy options."""
     overloads = []
-    ENGINES = [None, "dataset"]
+    ENGINES = ["xarray"]
```
Comment (hoxbro) [inline scripts/generate_typehints.py:49]: The only reason I called this dataset was also to make it possible to have a function return `xr.DataArray`, but I think this is fine. 

You also need to add this to the top to have it generate the type hints for the new datasets. 

We could add a test that checks that all callables in `__all__` are added to the script.

File: src/hvsampledata/__init__.py [hunk 2]
```diff
@@ -59,6 +59,71 @@ def large_timeseries(
...
+    -----------
+    The NCEP/NCAR Reanalysis 1 project is using a state-of-the-art analysis/forecast
+    system to perform data assimilation using past data from 1948 to the present.
+
+    This dataset was created by temporally resampling the `air_temperature` dataset
+    made available by xarray-data, itself being a spatial and temporal subset of
+    the original data. It only includes the air temperature variable.
+
+    Temporal coverage:
+    - Every 6 hours, starting from 00:00
```
Comment (hoxbro) [inline src/hvsampledata/__init__.py:86]: Did you try to take non-adjacent time values?

File: pixi.toml [hunk 3]
```diff
@@ -84,6 +84,7 @@ polars = "*"
 pyarrow = "*"
 rioxarray = "*"
 xarray = "*"
+netcdf4 = "*"
```
Comment (hoxbro) [inline pixi.toml:87]: Do you think we should warn/error early if this is not installed when calling the dataset? 

I'm unsure if netcdf4 is the only xarray engine to load the dataset. I haven't tried it, but Scipy may also work.

File: scripts/generate_typehints.py (hunk 1 above)
Comment (maximlt) [reply scripts/generate_typehints.py:49]: No I think it was a bug,`xarray` is an `engine`, `dataset` and `dataarray` are a `format`.

File: src/hvsampledata/__init__.py (hunk 2 above)
Comment (maximlt) [reply src/hvsampledata/__init__.py:86]: Do you mean daily?

Comment (hoxbro) [reply src/hvsampledata/__init__.py:86]: Yes or maybe weekly.

File: scripts/generate_typehints.py (hunk 1 above)
Comment (hoxbro) [reply scripts/generate_typehints.py:49]: I didn't put that much thought into the gridded dataset, so you are welcome to change things.

File: pixi.toml (hunk 3 above)
Comment (maximlt) [reply pixi.toml:87]: I'm not sure. I would like to add `extras` to the package that are documented and tested, with their minimum pins too.

File: src/hvsampledata/__init__.py (hunk 2 above)
Comment (maximlt) [reply src/hvsampledata/__init__.py:86]: No I didn't try.

Comment (hoxbro) [reply src/hvsampledata/__init__.py:86]: I mainly asked because I was curious and could see you had done some work to detect an interesting period.

If you feel this is a good dataset, I have no objections. :+1:

File: scripts/generate_typehints.py (hunk 1 above)
Comment (hoxbro) [reply scripts/generate_typehints.py:49]: So it doesn't get lost:

> You also need to add this to the ~~top~~ to have it generate the type hints for the new datasets.

https://github.com/holoviz/hvsampledata/blob/0feb75f6dbe1af8eb37e9bc7ddf8cd46fc344318/scripts/generate_typehints.py#L105

File: src/hvsampledata/__init__.py (hunk 2 above)
Comment (maximlt) [reply src/hvsampledata/__init__.py:86]: I guess we'll stick to how it is now, I didn't think about turning it into a daily dataset since the original data (not just in xarray) is every 6 hours.

**Review by hoxbro — APPROVED** [review]
